ecosystem/typescript/sdk_v2/tests/unit/bcs-helper.test.ts

```diff
@@ -0,0 +1,588 @@
+    const deserializer = new Deserializer(complexSerializable.bcsToBytes());
+    const deserializedComplexSerializable = ComplexSerializable.deserialize(deserializer);
+    expect(deserializedComplexSerializable.myU8.value).toEqual(complexSerializable.myU8.value);
+    expect(deserializedComplexSerializable.myU16.value).toEqual(complexSerializable.myU16.value);
+    expect(deserializedComplexSerializable.myU32.value).toEqual(complexSerializable.myU32.value);
+    expect(deserializedComplexSerializable.myU64.value).toEqual(complexSerializable.myU64.value);
+    expect(deserializedComplexSerializable.myU128.value).toEqual(complexSerializable.myU128.value);
+    expect(deserializedComplexSerializable.myU256.value).toEqual(complexSerializable.myU256.value);
+    expect(deserializedComplexSerializable.myBool.value).toEqual(complexSerializable.myBool.value);
+    expect(deserializedComplexSerializable.myString.value).toEqual(complexSerializable.myString.value);
+    expect(deserializedComplexSerializable.myVectorBool.values).toEqual(complexSerializable.myVectorBool.values);
+    expect(deserializedComplexSerializable.myVectorU8.values).toEqual(complexSerializable.myVectorU8.values);
+    expect(deserializedComplexSerializable.myVectorU16.values).toEqual(complexSerializable.myVectorU16.values);
+    expect(deserializedComplexSerializable.myVectorU32.values).toEqual(complexSerializable.myVectorU32.values);
+    expect(deserializedComplexSerializable.myVectorU64.values).toEqual(complexSerializable.myVectorU64.values);
+    expect(deserializedComplexSerializable.myVectorU128.values).toEqual(complexSerializable.myVectorU128.values);
+    expect(deserializedComplexSerializable.myVectorU256.values).toEqual(complexSerializable.myVectorU256.values);
+    expect(deserializedComplexSerializable.myVectorString.values).toEqual(complexSerializable.myVectorString.values);
+    expect(deserializedComplexSerializable.myOptionBool.value!.value).toEqual(
+      complexSerializable.myOptionBool.value!.value,
+    );
+    expect(deserializedComplexSerializable.myOptionU64.value).toEqual(complexSerializable.myOptionU64.value);
+    expect(deserializedComplexSerializable.myOptionU64.value).toEqual(undefined);
+    expect(deserializedComplexSerializable.myOptionString.value!.value).toEqual(
+      complexSerializable.myOptionString.value!.value,
+    );
+  });
+
+  it("serializes a rotation capability offer struct correctly", () => {
+    class RotationCapabilityOfferProofChallengeV2 extends Serializable {
+      public readonly moduleAddress: AccountAddress = AccountAddress.ONE;
+      public readonly moduleName: string = "account";
+      public readonly structName: string = "RotationCapabilityOfferProofChallengeV2";
+      public readonly functionName: string = "offer_rotation_capability";
+
+      constructor(
+        public readonly chainId: number,
+        public readonly sequenceNumber: number,
```

Comment:
sequenceNumber is defined in the sdk as `string | bigint`, how come here is `number`?

Reply:
just copied it over from the existing class, think it was made a long time ago, I'll remove it for now and we can put it in an example